File: src/main/java/com/jss/osiris/modules/profile/service/TeamService.java
package com.jss.osiris.modules.profile.service;

import java.util.List;

import com.jss.osiris.modules.profile.model.Team;

public interface TeamService {
    public List<Team> getTeams();

    public Team getTeam(Integer id);

    public Team addOrUpdateTeam(Team team);

    public Team getTeamIdByCode(String code);
}

File: client/src/app/modules/profile/services/team.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { Team } from '../../profile/model/Team';

@Injectable({
  providedIn: 'root'
})
export class TeamService extends AppRestService<Team>{

  constructor(http: HttpClient) {
    super(http, "profile");
  }

  getTeams() {
    return this.getList(new HttpParams(), "teams");
  }
  
   addOrUpdateTeam(team: Team) {
    return this.addOrUpdate(new HttpParams(), "team", team, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
